taca/flowcell_parser/classes.py

```diff
@@ -0,0 +1,322 @@
+        if self.runparameters:
+            self.obj.update(self.runparameters.data)
+            if self.runparameters.recipe:
+                self.obj['run_setup']=self.runparameters.recipe
+        if self.samplesheet:
+            self.obj['samplesheet_csv']=self.samplesheet.data
+        if self.lanebarcodes:
+            self.obj['illumina']={}
+            self.obj['illumina']['Demultiplex_Stats']={}
+            self.obj['illumina']['Demultiplex_Stats']['Barcode_lane_statistics']=self.lanebarcodes.sample_data
+            self.obj['illumina']['Demultiplex_Stats']['Flowcell_stats']=self.lanebarcodes.flowcell_data
+        
+
+
+
+
+class XTenLaneBarcodeParser(object):
+    def __init__(self, path ):
+        if os.path.exists(path):
+            self.path=path
+            self.parse()
+        else:
+            raise os.error("XTen laneBarcode.html cannot be found at {0}".format(path))
+
+    def parse(self):
+        self.sample_data=[]
+        self.flowcell_data={}
+        with open(self.path) as htmlfile:
+            bsoup=BeautifulSoup(htmlfile)
+            flowcell_table=bsoup.find_all('table')[1]
+            lane_table=bsoup.find_all('table')[2]
+
+            
+            keys=[]
+            values=[]
+            for th in flowcell_table.find_all('th'):
+                keys.append(th.text)
+            for td in flowcell_table.find_all('td'):
+                values.append(td.text)
+
+            self.flowcell_data = dict(zip(keys, values))
+
+            keys=[]
+            rows=lane_table.find_all('tr')
+            for row in rows[1:]:
+            #I want to skip the first row
+                if len(row.find_all('th')):
+                    #this is the header row
+                    seen_clusters=False
+                    for th in row.find_all('th'):
+                        key=th.text.replace('<br/>', ' ').replace('&gt;', '>')
+                        if key == '#':
+                            key='Lane'
+                        elif key == 'Clusters':
+                            if not seen_clusters:
+                                key= 'Raw Clusters'
+                                seen_clusters=True
+
+
+                        keys.append(key)
+                elif len(row.find_all('td')):
+                    values=[]
+                    for td in row.find_all('td'):
+                        values.append(td.text)
+
+                    d=dict(zip(keys,values))
+                    self.sample_data.append(d)
+
+
+
+
+class XTenDemultiplexingStatsParser(object):
+    def __init__(self, path ):
+        if os.path.exists(path):
+            self.path=path
+            self.parse()
+        else:
+            raise os.error("XTen DemultiplexingStats.xml cannot be found at {0}".format(path))
+
+    def parse(self):
+        data={}
+        tree=ET.parse(self.path)
+        root = tree.getroot()
+        self.data=xml_to_dict(root)
+
+
+class XTenSampleSheetParser(object):
```

Comment:
hmmm @vezzi coded some functions for parsing samplesheets as well [here](https://github.com/SciLifeLab/TACA/blob/master/taca/analysis/analysis.py#L128) have you guys checked with each other that you're not doing the same thing?